tests/ifs/ifs.c

```diff
@@ -110,25 +182,38 @@ static int scan_init(struct test *test)
                 ifs0 = open(PATH_SYS_IFS_BASE "intel_ifs_0", O_DIRECTORY | O_PATH | O_CLOEXEC);
         }
 
-
-        if (ifs0 < 0 || faccessat(ifs0, "run_test", R_OK, 0) < 0 ||
-                faccessat(ifs0, "status", R_OK, 0) < 0 || faccessat(ifs0, "details", R_OK, 0) < 0) {
-                log_info("not supported (missing kernel module, firmware, or invalid HW)");
-                close(ifs0);
-                return -EOPNOTSUPP;
-        }
-
-        /* see if we can open run_test for writing */
-        int fd = openat(ifs0, "run_test", O_WRONLY);
+        /* see if we can open run_test and current_batch for writing */
+        int run_fd = openat(ifs0, "run_test", O_WRONLY);
         int saved_errno = errno;
-        close(fd);
-        close(ifs0);
-        if (fd < 0) {
+        if (run_fd < 0) {
                 log_info("could not open intel_ifs_0/run_test for writing (not running as root?): %m");
-                close(fd);
+                close(run_fd);
                 return -saved_errno;
         }
 
+        int batch_fd = openat(ifs0, "current_batch", O_WRONLY);
+        saved_errno = errno;
+        if (batch_fd < 0) {
+                log_info("could not open intel_ifs_0/current_batch for writing (not running as root?): %m");
+                close(batch_fd);
+                return -saved_errno;
+        }
+        close(run_fd);
+        close(batch_fd);
+
+
+        /* load test file */
+        if (!load_test_file(ifs0, test, ifs_info))
```

Comment:
I'd say that since you've opened the batch file, you should simply pass the file descriptor here. You'd need to split `read_file` into two functions, one `read_file_fd` and the other that opens. There's no need for `write_file_fd`, that's just `write`